src/utils/ipfs/restIpfsNode.ts

```diff
@@ -0,0 +1,41 @@
+import axios from 'axios';
+
+export class RestIpfsNode {
+  readonly node_url: string;
+
+  private async fetch(
+    method: 'GET' | 'POST',
+    path: string,
+    data?: any
+  ): Promise<any> {
+    const url = `${this.node_url}${path}`;
+    try {
+      const response = await axios({
+        method,
+        url,
+        data,
+      });
+      return response.data;
+    } catch (error) {
+      console.log(error);
+      return null;
```

Comment:
here no error handling, need to pass error

Reply:
that can break current logic, will be refactored later